homeassistant/components/sensor/openweathermap.py

```diff
@@ -0,0 +1,193 @@
+"""
+homeassistant.components.sensor.openweathermap
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+OpenWeatherMap (OWM) service.
+
+Configuration:
+
+To use the OpenWeatherMap sensor you will need to add something like the
+following to your config/configuration.yaml
+
+sensor:
+  platform: openweathermap
+  api_key: YOUR_APP_KEY
+  monitored_variables:
+    - type: 'weather'
+    - type: 'temperature'
+    - type: 'wind_speed'
+    - type: 'humidity'
+    - type: 'pressure'
+    - type: 'clouds'
+    - type: 'rain'
+    - type: 'snow'
+
+VARIABLES:
+
+api_key
+*Required
+To retrieve this value log into your account at http://openweathermap.org/
+
+monitored_variables
+*Required
+An array specifying the variables to monitor.
+
+These are the variables for the monitored_variables array:
+
+type
+*Required
+The variable you wish to monitor, see the configuration example above for a
+list of all available variables
+
+Details for the API : http://bugs.openweathermap.org/projects/api/wiki
+
+Only metric measurement are supported at the moment.
+  # Location is Berne Switzerland
+  latitude: 46.9167
+  longitude: 7.4667
+
+  # C for Celcius, F for Fahrenheit
+  temperature_unit: C
```

Comment:
Why this option if it is metric only?

Also, don't worry about being metric only. Home Assistant will convert metric units to imperial if the user has set its preferences to imperial.